### PR TITLE
PLAT-355: fix check of message with token

### PR DESCRIPTION
**Pull request by ivanshibitov**
…current node 2. add more logging

**- What I did**
1. fix check of message with token: now compare sender with current node 
2. add more logging

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: ledger-core/application/testwalletapi/handlers.go [hunk 1]
```diff
@@ -80,7 +80,12 @@ func (s *TestWalletServer) Create(w http.ResponseWriter, req *http.Request) {
 		TraceID:   traceID,
 		Error:     "",
 	}
-	defer func() { s.mustWriteResult(w, result) }()
+	defer func() {
+		if len(result.Error) != 0 {
+			logger.Error(result.Error)
```
Comment (sexyhamster) [inline ledger-core/application/testwalletapi/handlers.go:85]: better to have whole result here, or at lest traceID

File: ledger-core/application/testwalletapi/handlers.go [hunk 2]
```diff
@@ -92,7 +97,7 @@ func (s *TestWalletServer) Create(w http.ResponseWriter, req *http.Request) {
 
 	walletRes, err := s.runWalletRequest(ctx, walletReq)
 	if err != nil {
-		result.Error = throw.W(err, "Failed to process create wallet contract call request").Error()
+		result.Error = throw.W(err, "Failed to process create wallet contract call request (Create)").Error()
```
Comment (sexyhamster) [inline ledger-core/application/testwalletapi/handlers.go:100]: why do you need extra (Create) here?

File: ledger-core/network/messagesender/service.go [hunk 3]
```diff
@@ -78,6 +78,10 @@ func (dm *DefaultService) SendRole(ctx context.Context, msg payload.Marshaler, r
 		return throw.W(err, "failed to calculate role")
 	}
 
+	if nodes[0].Equal(dm.affinity.Me()) {
+		inslogger.FromContext(ctx).Debug("Send to myself")
```
Comment (sexyhamster) [inline ledger-core/network/messagesender/service.go:82]: does this helps? if you are sending on the same machine, the log with trace will continue on the same machine

File: ledger-core/virtual/authentication/service.go [hunk 4]
```diff
@@ -66,12 +66,12 @@ func (s service) checkDelegationToken(expectedVE reference.Global, token payload
...
 
-	if token.Approver.Equal(s.affinity.Me()) {
-		return throw.New("selfNode cannot be equal to token Approver",
+	if sender.Equal(s.affinity.Me()) {
+		return throw.New("current node cannot be equal to sender of message with token",
 			struct {
 				SelfNode string
-				Approver string
-			}{SelfNode: s.affinity.Me().String(), Approver: token.Approver.String()})
+				Sender   string
```
Comment (sexyhamster) [inline ledger-core/virtual/authentication/service.go:73]: if they are equal you can log only one of them

File: ledger-core/virtual/handlers/handle_meta.go [hunk 5]
```diff
@@ -74,7 +74,14 @@ func (f FactoryMeta) Process(msg *statemachine.DispatcherMessage, pr pulse.Range
 		logger.Warn(throw.W(err, "illegitimate msg", struct {
 			messageTypeID uint64
 			messageType   reflect.Type
-		}{messageTypeID: payloadTypeID, messageType: payloadType}))
+			incomingPulse string
+			currentPulse  string
+		}{
+			messageTypeID: payloadTypeID,
+			messageType:   payloadType,
+			incomingPulse: payloadMeta.Pulse.String(),
```
Comment (sexyhamster) [inline ledger-core/virtual/handlers/handle_meta.go:82]: why string? pulse is number

File: ledger-core/application/testwalletapi/handlers.go (hunk 1 above)
Comment (ivanshibitov) [reply ledger-core/application/testwalletapi/handlers.go:85]: Logger contains traceId field
It's error path. There is no result here

File: ledger-core/network/messagesender/service.go (hunk 3 above)
Comment (ivanshibitov) [reply ledger-core/network/messagesender/service.go:82]: Multiple messages might be send with the same traceid. It helps to me for debugging.

File: ledger-core/application/testwalletapi/statemachine/statemachine.go [hunk 6]
```diff
@@ -63,6 +63,7 @@ func (s *SMTestAPICall) GetStateMachineDeclaration() smachine.StateMachineDeclar
 }
 
 func (s *SMTestAPICall) migrationDefault(ctx smachine.MigrationContext) smachine.StateUpdate {
+	ctx.Log().Trace("Resend message")
```
Comment (ruz) [inline ledger-core/application/testwalletapi/statemachine/statemachine.go:66]: please delete this trace, convlogtxt logs every step of SM. I don't understand why you need this?

Comment (ivanshibitov) [reply ledger-core/application/testwalletapi/statemachine/statemachine.go:66]: It adds more clarity when you need to troubleshoot

Comment (ruz) [reply ledger-core/application/testwalletapi/statemachine/statemachine.go:66]: I disagree, every step and every jump is logged in a defined format. I also add some troubleshooting points with spacers ("\n\n"), but don't leave them in commits.

**Pull request merged**